src/data-mgt/python/jobs/events.py

```diff
@@ -1,95 +1,45 @@
-import copy
 import json
 import os
-from threading import Thread
 import traceback
 import requests
 
 DEVICE_REGISTRY_STAGING_URL = os.getenv("DEVICE_REGISTRY_STAGING_URL")
 DEVICE_REGISTRY_PRODUCTION_URL = os.getenv("DEVICE_REGISTRY_PRODUCTION_URL")
 
 
-def measurements_insertion(data, tenant):
+class DeviceRegistry:
 
-    threads = []
+    def __init__(self, measurements, tenant, device_name):
+        self.__measurements = measurements
+        self.__tenant = tenant
+        self.__device_name = device_name
 
-    staging_data = copy.deepcopy(data)
+    def insert_measurements(self):
 
-    # production
-    thread_production = Thread(target=production_insertion, args=(data, tenant,))
-    threads.append(thread_production)
-    thread_production.start()
+        self.__add_to_events_collection(DEVICE_REGISTRY_PRODUCTION_URL)
+        self.__add_to_events_collection(DEVICE_REGISTRY_STAGING_URL)
 
-    # staging
-    thread_staging = Thread(target=staging_insertion, args=(staging_data, tenant,))
-    threads.append(thread_staging)
-    thread_staging.start()
+    def __add_to_events_collection(self, base_url):
 
-    # wait for all threads to terminate
-    for thread in threads:
-        thread.join()
+        try:
 
+            json_data = json.dumps(self.__measurements)
 
-def staging_insertion(data, tenant):
+            headers = {'Content-Type': 'application/json'}
 
-    """
-    sends device measurements to device registry microservice on staging environment
-    :param data: device measurements (includes device name to reduce on function args)
-    :param tenant: organisation eg airqo
-    :return: none
-    """
+            base_url = base_url + "devices/events/add?device=" + self.__device_name + "&tenant=" + self.__tenant
```

Comment:
in the future you can explore the use of python `f-string` to concatenate strings

Reply:
Thanks, let me make the update